File: Minimum_Number_of_Steps_to_Make_Two_Strings_Anagram_II/Minimum_Number_of_Steps_to_Make_Two_Strings_Anagram_II.py
s = "leetcode"
t = "coats"

def minSteps(s,t):
    m = {}
    for i in s:
        if i in m:
            m[i] +=1
        else:
            m[i] = 1

    for t1 in t:
        if t1 in m:
            m[t1] -= 1
        else:
            m[t1] = -1

    res=0
    for i,j in m.items():
        res+=abs(j)

    return res


print(minSteps(s,t))

File: Minimum_Number_of_Steps_to_Make_Two_Strings_Anagram_II/Minimum_Number_of_Steps_to_Make_Two_Strings_Anagram_II.go
package main

import (
	"fmt"
	"math"
)

func main() {
	s, t := "leetcode", "coats"
	fmt.Println(minSteps(s, t))
}

func minSteps(s, t string) int {
	mk := make(map[string]int)
	for _, s1 := range s {
		mk[string(s1)]++
	}

	for _, t1 := range t {
		mk[string(t1)]--
	}
	res := 0
	for _, i := range mk {
		res += int(math.Abs(float64(i)))
	}

	return res
}
